crates/rpc-types-engine/src/payload/mod.rs

```diff
@@ -19,3 +22,53 @@ pub enum OpExecutionPayload {
     /// V4 payload
     V4(OpExecutionPayloadV4),
 }
+
+impl OpExecutionPayload {
+    /// Returns a reference to the V2 payload, if any.
+    pub const fn as_v2(&self) -> Option<&ExecutionPayloadV2> {
+        match self {
+            Self::V2(payload) => Some(payload),
+            Self::V3(payload) => Some(&payload.payload_inner),
+            Self::V4(payload) => todo!(),
+        }
+    }
+
+    /// Returns a mutable reference to the V2 payload, if any.
+    pub fn as_v2_mut(&mut self) -> Option<&mut ExecutionPayloadV2> {
+        match self {
+            Self::V2(payload) => Some(payload),
+            Self::V3(payload) => Some(&mut payload.payload_inner),
+            Self::V4(payload) => todo!(),
+        }
+    }
+    /// Returns the parent hash for the payload.
+    pub const fn parent_hash(&self) -> B256 {
+        self.as_v2().unwrap().payload_inner.parent_hash
+    }
+
+    /// Returns the block hash for the payload.
+    pub const fn block_hash(&self) -> B256 {
+        self.as_v2().unwrap().payload_inner.block_hash
+    }
+
+    /// Returns the block number for this payload.
+    pub const fn block_number(&self) -> u64 {
+        self.as_v2().unwrap().payload_inner.block_number
+    }
+
+    /// Converts [`ExecutionPayloadV1`] to [`Block`].
+    ///
+    /// Caution: This does not set fields that are not part of the payload and only part of the
+    /// [`ExecutionPayloadSidecar`]:
+    /// - parent_beacon_block_root
+    /// - requests_hash
+    ///
+    /// See also: [`ExecutionPayload::try_into_block_with_sidecar`]
```

Comment:
```suggestion
    /// See also: [`OpExecutionPayload::try_into_block_with_sidecar`]
```
we need this method to be impl as well as update docs here